cli/helpers.ts

```diff
@@ -0,0 +1,216 @@
+import * as prettier from "prettier";
+import * as fs from "fs";
+import * as fse from "fs-extra";
+import * as path from "path";
+import chalk from "chalk";
+export const log = console.log;
+import getPort from "get-port";
+
+export enum Package {
+  Frontend = "frontend",
+  Backend = "backend",
+  TopLevel = "toplevel",
+}
+
+export class Ports {
+  frontend: number | undefined = undefined;
+  backend: number | undefined = undefined;
+  db: number | undefined = undefined;
+  setup = async (): Promise<void> => {
+    if (this.backend === undefined) {
+      this.backend = await getPort();
+    }
+    if (this.frontend === undefined) {
+      this.frontend = await getPort();
+    }
+    if (this.db === undefined) {
+      this.db = await getPort();
+    }
+  };
+}
+
+export const FRONTEND_REGEXP = /\NEXT_PUBLIC_GRAPHQL_URL=http:\/\/localhost:\d+\/graphql/;
+export const BACKEND_REGEXP = /\PORT=\d+/g;
+export const DATABASE_URL_REGEXP = /\DATABASE_URL=postgres:\/\/postgres:password@localhost\/postgres/;
+
+export class UnreachableCaseError extends Error {
+  constructor(val: never) {
+    super(`Unreachable case: ${JSON.stringify(val)}`);
+  }
+}
+
+export function copyTemplate(srcDir: string, targetDir: string) {
+  // Copy template, ignoring node_modules
+  try {
+    fse.copySync(srcDir, targetDir, {
+      filter: function (path) {
+        return !(path.indexOf("node_modules") > -1);
+      },
+    });
+    log(chalk.cyan(`Template copied to:\n  ${chalk.cyan.bold(targetDir)}`));
+  } catch {
+    throw new Error(
+      chalk.red(`Error copying template from ${srcDir} to ${targetDir}`)
+    );
+  }
+}
+
+export function updateTemplate(
+  appName: string,
+  targetDir: string,
+  ports: Ports
+) {
+  editReadme(appName, targetDir);
+  editPackageJson(appName, targetDir, Package.Frontend, ports);
+  editPackageJson(appName, targetDir, Package.Backend, ports);
+  editPackageJson(appName, targetDir, Package.TopLevel, ports);
+  editEnvFile(targetDir, Package.Frontend, ports);
+  editEnvFile(targetDir, Package.Backend, ports);
+  log(
+    chalk.cyan(
+      `Updated template with values: 
+  package name: ${chalk.cyan.bold(appName)}
+  frontend port: ${chalk.cyan.bold(ports.frontend)}
+  backend port: ${chalk.cyan.bold(ports.backend)}
+  database port: ${chalk.cyan.bold(ports.db)}
+`
+    )
+  );
+}
+
+export function editReadme(appName: string, targetDir: string) {
+  const filename = path.join(targetDir, `README.md`);
+
+  try {
+    const readmeContents = `#${appName}\n\nGenerated with create-originate-app`;
+    fs.writeFileSync(filename, readmeContents);
+    log(chalk.blue(`Prepared ${filename}`));
+  } catch (err) {
+    throw new Error(chalk.red(`Error preparing ${filename}\n${err}`));
+  }
+}
+
+export function editPackageJson(
+  appName: string,
+  targetDir: string,
+  packageName: Package,
+  ports: Ports
+) {
+  let filename!: string;
```

Comment:
I'd rather check whether this value is defined later instead of using a non-null assertion.